core/blockchain.go

```diff
@@ -1736,3 +1743,47 @@ func (bc *BlockChain) SubscribeLogsEvent(ch chan<- []*types.Log) event.Subscript
 func (bc *BlockChain) SubscribeBlockProcessingEvent(ch chan<- bool) event.Subscription {
 	return bc.scope.Track(bc.blockProcFeed.Subscribe(ch))
 }
+
+// GetAncestorHash return the hash of ancestor at the given number
+func (bc *BlockChain) GetAncestorHash(ref *types.Header, target uint64) common.Hash {
```

Comment:
Is there a reason not to put this logic in `HeaderChain`?

Reply:
No, it should probably wind up there instead. Right now, however, it's only used for EVM execution of `BLOCKHASH`, and I was more confident in how the blockchain works than how the headerchain works. 

If this is becomes used from any other place, like p2p, then it should definitely go into headerchain instead.

Reply:
Makes sense. Thanks for clarifying.